src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -57,92 +58,108 @@ using Transport::Session;
 
-CHIP_ERROR GeneralCommissioningAttrAccess::ReadSupportsConcurrentConnection(AttributeValueEncoder & aEncoder)
-{
-    SupportsConcurrentConnection::TypeInfo::Type supportsConcurrentConnection;
+            // TODO: The commissioner might use the critical parameters in BasicCommissioningInfo to initialize
+            // the CommissioningParameters at the beginning of commissioning flow.
+            basicCommissioningInfo.failSafeExpiryLengthSeconds  = CHIP_DEVICE_CONFIG_FAILSAFE_EXPIRY_LENGTH_SEC;
+            basicCommissioningInfo.maxCumulativeFailsafeSeconds = CHIP_DEVICE_CONFIG_MAX_CUMULATIVE_FAILSAFE_SEC;
+            static_assert(CHIP_DEVICE_CONFIG_MAX_CUMULATIVE_FAILSAFE_SEC >= CHIP_DEVICE_CONFIG_FAILSAFE_EXPIRY_LENGTH_SEC,
+                          "Max cumulative failsafe seconds must be larger than failsafe expiry length seconds");
 
-    // TODO: The commissioner might use the critical parameters in BasicCommissioningInfo to initialize
-    // the CommissioningParameters at the beginning of commissioning flow.
-    supportsConcurrentConnection = (CHIP_DEVICE_CONFIG_SUPPORTS_CONCURRENT_CONNECTION) != 0;
+            return aEncoder.Encode(basicCommissioningInfo);
+        }
+        case SupportsConcurrentConnection::Id: {
+            SupportsConcurrentConnection::TypeInfo::Type supportsConcurrentConnection;
 
-    return aEncoder.Encode(supportsConcurrentConnection);
-}
+            // TODO: The commissioner might use the critical parameters in BasicCommissioningInfo to initialize
+            // the CommissioningParameters at the beginning of commissioning flow.
+            supportsConcurrentConnection = (CHIP_DEVICE_CONFIG_SUPPORTS_CONCURRENT_CONNECTION) != 0;
+
+            return aEncoder.Encode(supportsConcurrentConnection);
+        }
+#if CHIP_CONFIG_TC_REQUIRED
+        case TCAcceptedVersion::Id: {
+            auto provider = Server::GetInstance().GetEnhancedSetupFlowProvider();
+            auto getter   = &EnhancedSetupFlowProvider::GetTermsAndConditionsAcceptedAcknowledgementsVersion;
+            return ReadIfSupported(provider, getter, aEncoder);
+        }
+        case TCMinRequiredVersion::Id: {
+            auto provider = Server::GetInstance().GetEnhancedSetupFlowProvider();
+            auto getter   = &EnhancedSetupFlowProvider::GetTermsAndConditionsRequiredAcknowledgementsVersion;
+            return ReadIfSupported(provider, getter, aEncoder);
+        }
+        case TCAcknowledgements::Id: {
+            auto provider = Server::GetInstance().GetEnhancedSetupFlowProvider();
+            auto getter   = &EnhancedSetupFlowProvider::GetTermsAndConditionsAcceptedAcknowledgements;
+            return ReadIfSupported(provider, getter, aEncoder);
+        }
+        case TCAcknowledgementsRequired::Id: {
+            auto provider = Server::GetInstance().GetEnhancedSetupFlowProvider();
+            auto getter   = &EnhancedSetupFlowProvider::GetTermsAndConditionsRequiredAcknowledgements;
```

Comment:
The implementation for reading TCAcknowledgementsRequired attribute does not seem to match the spec.
I understand this attribute supposed to be a boolean value indicate whether SetTCAcknowledgements is required to be called.
On the other hand, the uint16 value (represents TC required to be accepted) can be get with this GetTermsAndConditionsRequiredAcknowledgements function.